include/podio/UserDataCollection.h

```diff
@@ -77,6 +77,12 @@ class UserDataCollection : public CollectionBase {
   VectorMembersInfo m_vecmem_info{};
 
 public:
+  using value_type = typename decltype(_vec)::value_type;
```

Comment:
```suggestion
  using value_type = typename std::vector<BasicType>::value_type;
```

Or alternatively introduce a `using VecType = std::vector<BasicType>` (private) member type, to not have the `decltype` in there. I think that would make this a bit more readable.